Packs/HackerOne/Integrations/HackerOne/HackerOne.py

```diff
@@ -0,0 +1,588 @@
+        hr["Severity"] = severity.get("rating", "")
+        hr["Created At"] = attributes.get("created_at")
+        hr["Vulnerability Information"] = attributes.get("vulnerability_information")
+        reporter = relationships.get("reporter", {})
+        relationship_data = reporter.get("data", {})
+        inner_attributes = relationship_data.get("attributes", {})
+        hr["Reporter Username"] = inner_attributes.get("username")
+
+        reports_hr.append(hr)
+    return tableToMarkdown("Report(s)", reports_hr,
+                           headers=["Report ID", "Reporter Username", "Title", "State", "Severity", "Created At",
+                                    "Vulnerability Information"], removeNull=True)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises:
+     exceptions if something goes wrong.
+
+    Args:
+        client: client to use
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    client.program_list(params={"page[size]": 1})
+
+    if demisto.params().get('isFetch'):
+        fetch_incidents(client, {})
+
+    return 'ok'
+
+
+def fetch_incidents(client: Client, last_run: dict, ) -> Tuple[dict, list]:
+    """Fetches incidents from HackerOne.
+
+    :type client: ``Client``
+    :param client: client to use
+
+    :type last_run: ``Dict[str, str]``
+    :param last_run: Last run returned by function demisto.getLastRun
+
+    :type params: ``Dict[str, str]``
+    :param params: Arguments for fetch-incident.
+
+    :rtype: ``Tuple``
+    :return: tuple of dictionary of next run and list of fetched incidents
+    """
+    validate_fetch_incidents_parameters(client.max_fetch, client.program_handle, client.filters)
+    fetch_params = prepare_fetch_incidents_parameters(client.max_fetch, client.first_fetch, client.program_handle,
+                                                      client.severity, client.state, client.filters)
+    first_fetch = fetch_params["filter[created_at__gt]"]
+
+    fetch_params["page[number]"] = last_run.get("next_page", 1)
+    fetch_params["filter[created_at__gt]"] = last_run.get("current_created_at", first_fetch)  # type: ignore
+
+    response = client.report_list(params=fetch_params)
+
+    results = response.get('data', [])
+
+    if not results:
+        next_run = last_run
+        next_run["current_created_at"] = last_run.get("next_created_at", first_fetch)
+        next_run["next_page"] = 1
+        return next_run, []
+
+    next_run = {"next_page": fetch_params["page[number]"] + 1,
+                "current_created_at": fetch_params["filter[created_at__gt]"],
+                "next_created_at": results[-1].get("attributes", {}).get("created_at")}
```

Comment:
Why do we need all this data, specifically the next_page and current_created_at? Why can't we only store the time of the last incident that we fetched and get incidents that have greater time?

Reply:
Consider the situation where there are multiple records with the same datetime and half of them are on the following page. Our reports will be lost if we do not store the next page. This issue was discovered during quality assurance testing and as a result, we're storing these values.

Reply:
Let's go over this on tomorrow's call